mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -248,6 +258,26 @@ public void close() throws IOException {
       return iterable;
     }
 
+    @SuppressWarnings("unchecked")
+    private CloseableIterable<T> open(FileScanTask currentTask, Schema readSchema) {
+      CloseableIterable<T> iter;
+      switch (inMemoryDataModel) {
+        case PIG:
+        case HIVE:
+          // TODO implement value readers for Pig and Hive
+          throw new UnsupportedOperationException("Avro support not yet supported for Pig and Hive");
+        case GENERIC:
+          DeleteFilter deletes = new GenericDeleteFilter(io, currentTask, tableSchema, readSchema);
+          Schema requiredSchema = deletes.requiredSchema();
+          iter = deletes.filter(openTask(currentTask, requiredSchema));
```

Comment:
Why not return `deletes.filter(...)` here? That would remove the need for `iter` and `break`.

Reply:
Updated.